cli/src/commands.rs

```diff
@@ -109,16 +118,24 @@ pub enum Commands {
     /// `status` subcommand
     #[options(help = "show yubikey status")]
     Status(StatusCmd),
+
+    //TODO limit number of options to only valid slot IDs
+    /// `readcertificate` subcommand
+    #[options(help = "read certificate from designated slot")]
+    Readcertificate(ReadCertificateCmd),
```

Comment:
I was thinking these could be broken up into subcommands by the object they work on, so, instead of `readcertificate`, this would just be `enum Certificate`, which contains a `Certificate::Read` subcommand

Reply:
OK. I can give a look at that. How about passing options from the top level? I'd changed the signature on a function you'd written to pass in the CLI struct.